src/test/java/com/example/cats/service/CosmoCatServiceTest.java

```diff
@@ -0,0 +1,65 @@
+package com.example.cats.service;
+
+import com.example.cats.featuretoggle.FeatureToggleService;
+import com.example.cats.featuretoggle.FeatureToggles;
+import com.example.cats.featuretoggle.aspect.FeatureToggleAspect;
+import com.example.cats.featuretoggle.exception.FeatureToggleNotEnabledException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.aop.aspectj.annotation.AspectJProxyFactory;
+import java.util.List;
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayName("CosmoCatService Unit Tests (AOP)")
+class CosmoCatServiceTest {
+
+    private CosmoCatService cosmoCatService;
+    private CosmoCatService proxiedService;
+
+    @Mock
+    private FeatureToggleService featureToggleService;
+
+    private FeatureToggleAspect featureToggleAspect;
+
+    @BeforeEach
+    void setUp() {
+        cosmoCatService = new CosmoCatService();
```

Comment:
https://youtu.be/xcw5veP0TPw?si=3gFOmmgx2QZue1nr - that's not good idea in general you polute the memory for nothing. And besicly as well win nothing from this. But that's for different lab, so I won't minus here, but on other